packages/clay-list/src/ListWithItems.tsx

```diff
@@ -0,0 +1,162 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+import classNames from 'classnames';
+import ClayIcon from '@clayui/icon';
+import ClayLabel from '@clayui/label';
+import ClayList from './List';
+import ClaySticker from '@clayui/sticker';
+import React from 'react';
+import {ClayCheckbox} from '@clayui/form';
+import {ClayDropDownWithBasicItems} from '@clayui/drop-down';
+
+interface IListItem {
+	description?: string;
+	dropdownActions?: Array<any>;
+	href?: string;
+	labels?: Array<any>;
+	quickActions?: Array<any>;
+	symbol?: string;
+	title?: string;
+	[key: string]: any;
+}
+
+interface IProps extends React.HTMLAttributes<HTMLDivElement> {
+	header?: string;
+	identifier?: string;
+	items?: Array<IListItem>;
+	onSelectedItemsChange?: (items: Array<IListItem>) => void;
+	spritemap?: string;
+}
+
+interface IBooleanMap {
+	[s: string]: boolean;
+}
+
+export const ClayListWithItems: React.FunctionComponent<IProps> = ({
+	className,
+	header,
+	identifier,
+	items = [],
+	onSelectedItemsChange = () => {},
+	spritemap,
+	...otherProps
+}) => {
+	const defaultMapValue = React.useMemo(
+		() =>
+			items.reduce((acc, item, i) => {
+				acc[identifier ? item[identifier] : i] = false;
+				return acc;
+			}, {}),
+		[]
+	);
+
+	const [selectedMap, setSelectedMap] = React.useState<IBooleanMap>(
+		defaultMapValue
+	);
+
+	return (
+		<div {...otherProps} className={classNames(className)}>
+			<ClayList>
+				<ClayList.Header>{header}</ClayList.Header>
+
+				{items.map((item, i) => {
+					const keyVal = identifier ? item[identifier] : i;
+
+					const checked = selectedMap[keyVal];
+
+					const onItemSelect = () => {
+						const newItems: IBooleanMap = {
+							...selectedMap,
+							[keyVal]: !checked,
+						};
+
+						setSelectedMap(newItems);
+
+						onSelectedItemsChange(
+							items.filter(
+								(item, j) =>
+									newItems[identifier ? item[identifier] : j]
+							)
+						);
+					};
+
+					return (
+						<ClayList.Item active={checked} flex key={i}>
+							<ClayList.ItemField>
```

Comment:
We should make the items here optional, usually we had a `contentRender` to render different content but it was specific to Soy (deltemplates).

Reply:
Make the content of the items optional? I am following the lexicon design of a basic list and thought it'd be best to have pre-defined content for a high-level. Otherwise, this component starts to look identical to the low-level, except for a `map` over items.

Reply:
> Make the content of the items optional? I am following the lexicon design of a basic list and thought it'd be best to have pre-defined content for a high-level. Otherwise, this component starts to look identical to the low-level, except for a map over items.

I think we got into the same conversation we had on DropDown, but that might complement those conversations.

In the document I sent above it shows some variations and interactions on the rows (https://docs.google.com/document/d/1LWHKwKu3LQR6LlJGlXEv8qFzI1EgItYNDWJHnZmBGP0/edit#heading=h.rlrk2prxkmce).

I disagree about it becoming similar, in the end they can be visually equal but what we deliver at the low level is the great flexibility of people adding new things, changing, creating their own logic, using some low-level components... this It is the purpose of low levels. High levels do not need to be tied to a specific use case it can also be flexible, it would be the minimum of a component, otherwise you will not be able to use it elsewhere, just on some main pages. We need to offer flexibility here in some way, because we should not write the components as we would be writing just for our product for example...

I think this is a long conversation. I will try to better describe my thoughts than I am thinking to Clay and my perceptions of what we should deliver to low and high levels. It's my perception of the future I'm aiming at.

Reply:
Yeah this conversation probably starts to bleed over into a longer one. I think this can probably be continued in this issue I opened, https://github.com/liferay/clay/issues/2275.

We really need to iron out a defining guide on how open/closed we make high-level API. I can imagine this being a bottle neck for us going forward if we don't have proper definitions and guidelines for us to follow.

Reply:
Also, since its likely this conversation in that issue won't necessarily be resolved overnight. Do you want to postpone this PR or go ahead with the existing structure, and adapt changes later if needed?

Reply:
> We really need to iron out a defining guide on how open/closed we make high-level API. I can imagine this being a bottle neck for us going forward if we don't have proper definitions and guidelines for us to follow.

I agree, this is starting to be a bottleneck for us, we need to have a definition about it. I will spend some time writing what I think.

> Also, since its likely this conversation in that issue won't necessarily be resolved overnight. Do you want to postpone this PR or go ahead with the existing structure, and adapt changes later if needed?

I think we can go ahead with this and then we can apply it depending on what we set just to not delay this component anymore. Then we have to write documents for these new components that we add and detail that too because we need some arguments and definitions for when people come to ask us.